openverse_catalog/dags/providers/provider_api_scripts/freesound.py

```diff
@@ -8,294 +8,249 @@
-    "num_downloads,avg_rating,num_ratings,geotag,previews",
-}
-
-delayed_requester = DelayedRequester(DELAY)
-audio_store = AudioStore(provider=PROVIDER)
-
-logging.basicConfig(
-    format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s",
-    level=logging.INFO,
-)
 logger = logging.getLogger(__name__)
 
-preview_bitrates = {
-    "preview-hq-mp3": 128000,
-    "preview-lq-mp3": 64000,
-    "preview-hq-ogg": 192000,
-    "preview-lq-ogg": 80000,
-}
-
-
-def main(date="all"):
-    """This script pulls the data for a given date from the Freesound,
-    and writes it into a .TSV file to be eventually read
-    into our DB.
-
-    Required Arguments:
-
-    date:  Date String in the form YYYY-MM-DD.  This is the date for
-           which running the script will pull data.
-    """
-
-    logger.info("Begin: Freesound script")
-    licenses = ["Attribution", "Attribution Noncommercial", "Creative Commons 0"]
-    for license_name in licenses:
-        audio_count = _get_items(license_name, date)
-        logger.info(f"Audios for {license_name} pulled: {audio_count}")
-    logger.info("Terminated!")
-
-
-def _get_query_params(
-    license_name="",
-    page_number=1,
-    default_query_params=None,
-):
-    default_query_params = default_query_params or DEFAULT_QUERY_PARAMS
-    return {
-        **default_query_params,
-        "page": str(page_number),
-        "license": license_name,
+
+class FreesoundDataIngester(ProviderDataIngester):
+    batch_limit = 150
+    RETRIES = 3
+    host = "freesound.org"
+    endpoint = f"https://{host}/apiv2/search/text"
+    providers = {"audio": prov.FREESOUND_DEFAULT_PROVIDER}
+    flaky_exceptions = (SSLError, ConnectionError)
+    preview_bitrates = {
+        "preview-hq-mp3": 128000,
+        "preview-lq-mp3": 64000,
+        "preview-hq-ogg": 192000,
+        "preview-lq-ogg": 80000,
     }
 
+    HEADERS = {
+        "Accept": "application/json",
+    }
 
-def _get_items(license_name, date):
-    item_count = 0
-    page_number = 1
-    should_continue = True
-    default_query_params = copy.deepcopy(DEFAULT_QUERY_PARAMS)
-    try:
-        start_date = datetime.strftime(
-            datetime.fromisoformat(date), "%Y-%m-%dT%H:%M:%SZ"
-        )
-    except ValueError:
-        start_date = "*"
-    default_query_params["filter"] = f"created:[{start_date} TO NOW]"
-    while should_continue:
-        query_param = _get_query_params(
-            default_query_params=default_query_params,
-            license_name=license_name,
-            page_number=page_number,
-        )
-        batch_data = _get_batch_json(query_param=query_param)
-        if batch_data:
-            item_count = _process_item_batch(batch_data)
-            page_number += 1
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.api_key = Variable.get("API_KEY_FREESOUND")
+        self.headers = {"api_key": self.api_key}
```

Comment:
It looks like the api key is passed in through the query params for Freesound and does not need to be sent as a header. This also overrides the `"Accept": "application/json"`

Reply:
Oops! Thanks for catching this 😄